src/util/desktop-notifier.js

```diff
@@ -0,0 +1,22 @@
+/* @flow */
+import {NotificationCenter} from 'node-notifier';
+
+const defaultNotifier = new NotificationCenter({
+  withFallback: true,
+});
+
+type desktopNotificationsParams = {
+  title: string,
+  message: string,
+  icon?: string,
+  notifier?: typeof NotificationCenter,
```

Comment:
This shouldn't be part of the main params because it doesn't relate to the notification call itself. You should pass it in as a secondary option so that it won't ever collide with a real notification param (which is unlikely but this follows our pattern elsewhere in web-ext).

Example:
````javascript
export function desktopNotifications(
  {
    title, message, icon,
  }: desktopNotificationsParams,
  {
    notifier = defaultNotifier,
  }: desktopNotificationOptions = {}
): void {
````